File: 206-reverse-linked-list/206-reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* reverseList(struct ListNode* head){
  if(head == NULL)
    return NULL;
  if(head->next == NULL)
    return head;
  struct ListNode *front = reverseList(head->next);
  struct ListNode *back = head->next;
  back->next = head;
  head->next = NULL;
  return front;
}

File: 206-reverse-linked-list/206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
      if(head == None):
        return None
      if(head.next == None):
        return head
      back, front = head.next, self.reverseList(head.next)
      back.next = head
      head.next = None
      return front